bazel/dependency_imports.bzl

```diff
@@ -11,7 +11,7 @@ load("@rules_antlr//antlr:deps.bzl", "antlr_dependencies")
 load("@proxy_wasm_rust_sdk//bazel:dependencies.bzl", "proxy_wasm_rust_sdk_dependencies")
 
 # go version for rules_go
-GO_VERSION = "1.15.5"
+GO_VERSION = "1.16.6"
```

Comment:
Go 1.17 is now out (latest is Go 1.17.1: https://golang.org/dl/), any chance we can try updating to that? Hopefully a drop-in in this PR but can make it a follow-up PR if needed.

Reply:
Typically the `rules_go` release updates were very close to Go releases. That seems to have stopped with 0.28.0

Reply:
FWIW, my read of https://github.com/bazelbuild/rules_go/pull/2729 is that it should no longer be required to update `rules_go` for every new Go release.

Reply:
Interesting. We we need to continue to set the Go version to get reproducible builds (and to stop a potential breakage as the Go version increment in the background). So the changes to `rules_go` `0.28.0` and `bazel-gazelle` to `0.23.0` LGTM. We should bump the Go version on line 14 to `1.17.1` (I've been running this setup locally since `1.17` came out

Reply:
I did the minimum bump required to get M1 support at the time I submitted this PR just to keep this change as small as possible, happy to bump more, it will probably be fine